src/Microsoft.Health.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -23,12 +23,24 @@ public static class ServiceCollectionExtensions
         /// <param name="assembly">The assembly.</param>
         /// <param name="constructorParams">An array of constructor parameters that can be injected into IStartupModule on creation.</param>
         public static void RegisterAssemblyModules(this IServiceCollection collection, Assembly assembly, params object[] constructorParams)
+        {
+            collection.RegisterAssemblyModules(assembly: assembly, moduleTypeFilter: null, constructorParams: constructorParams);
+        }
+
+        /// <summary>
+        /// Registers assembly modules.
+        /// </summary>
+        /// <param name="collection">The collection.</param>
+        /// <param name="assembly">The assembly.</param>
+        /// <param name="moduleTypeFilter">Filter out modules that are not be registered.</param>
+        /// <param name="constructorParams">An array of constructor parameters that can be injected into IStartupModule on creation.</param>
+        public static void RegisterAssemblyModules(this IServiceCollection collection, Assembly assembly, Func<Type, bool> moduleTypeFilter, params object[] constructorParams)
```

Comment:
Another alternative: could we instead simply allow users to register specific modules, like:
```suggestion
        public static void RegisterModule<T>(this IServiceCollection collection, params object[] constructorParams) where T : IStartupModule
```